spec.bs

```diff
@@ -1440,8 +1440,22 @@ To <dfn export>generate a [=/pattern string=]</dfn> from a given [=/part list=]
       <ul>
         <li>|part|'s [=part/suffix=] is not the empty string.</li>
         <li>|part|'s [=part/prefix=] is not the empty string and is not |options|'s [=options/prefix code point=].</li>
-        <li>|custom name| is true, |part|'s [=part/type=] is "<a for=part/type>`segment-wildcard`</a>", |part|'s [=part/modifier=] is "<a for=part/modifier>`none`</a>", |next part| is not null, |next part|'s [=part/type=] is not "<a for=part/type>`fixed-text`</a>", |next part|'s [=part/prefix=] is the empty string, |next part|'s [=part/suffix=] is the empty string, and |next part|'s [=part/name=][0] is an [=ASCII digit=].
       </ul>
+    1. If all of the following are true:
+      <ul>
+        <li>|needs grouping| is false; and</li>
+        <li>|custom name| is true; and</li>
+        <li>|part|'s [=part/type=] is "<a for=part/type>`segment-wildcard`</a>"; and</li>
+        <li>|part|'s [=part/modifier=] is "<a for=part/modifier>`none`</a>"; and</li>
+        <li>|next part| is not null; and</li>
+        <li>|next part|'s [=part/prefix=] is the empty string; and</li>
+        <li>|next part|'s [=part/suffix=] is the empty string</li>
+      </ul>
+      then:
+      1. If |next part|'s [=part/type=] is "<a for=part/type>`fixed-text`</a>":
+        1. Set |needs grouping| to true if the result of running [=is a valid name code point=] given |next part|'s [=part/value=]'s first [=/code point=] and then boolean false is true.
```

Comment:
Stylistically I tend to prefer single-step `1. If X, then Y` instead of `1. If X:\n\t1. Y`, but that's a personal choice.

Reply:
I think in this case it becomes very hard to read since there is an additional conditional here.  Its really, "if X and Y, then foo" and then "if !X and Z, then bar".  Repeating the X part seems more verbose and harder to read to me.

Reply:
I was thinking:

```
1. If X:
  1. If not Y, Z
  1. Else if W, Q
```